modules/asm/scripts/install_asm.sh

```diff
@@ -25,31 +25,180 @@ PROJECT_ID=$1
 CLUSTER_NAME=$2
 CLUSTER_LOCATION=$3
 ASM_VERSION=$4
-MANAGED=$5
-MODE="install"
-
-# Download the correct version of the install_asm script
-curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm
-chmod u+x install_asm
-
-declare -a params=(
-    "--verbose"
-    "--project_id ${PROJECT_ID}"
-    "--cluster_name ${CLUSTER_NAME}"
-    "--cluster_location ${CLUSTER_LOCATION}"
-    "--mode ${MODE}"
-    "--enable_cluster_labels"
-    "--enable_cluster_roles"
-)
-
-# Add the --managed param if MANAGED is set to true
-if [[ "${MANAGED}" == true ]]; then
-    params+=("--managed")
-fi
-
-# Run the script with appropriate flags
-echo "Running ./install_asm" "${params[@]}"
-
-# Disable shell linting. Other forms will prevent the command to work
-# shellcheck disable=SC2046,SC2116
-./install_asm $(echo "${params[@]}")
+MODE=$5
+MCP=$6
+SKIP_VALIDATION=$7
+OPTIONS_LIST=$8
+CUSTOM_OVERLAYS_LIST=$9
+ENABLE_ALL=${10}
+ENABLE_CLUSTER_ROLES=${11}
+ENABLE_CLUSTER_LABELS=${12}
+ENABLE_GCP_APIS=${13}
+ENABLE_GCP_IAM_ROLES=${14}
+ENABLE_GCP_COMPONENTS=${15}
```

Comment:
Since there is an ask to enable these via module, I am thinking we should transform these flags into native Terraform resources within the module that does the same thing as the script. This way the provisioner is only used when there is a gap. Can be done in a follow up PR. 

/cc @morgante if you had any thoughts

Reply:
Agreed, we should only depend on the script for things which cannot be done natively in Terraform. I'm fine if it's swapped in a follow-up PR but we should hold releasing this until that is done (to avoid needing a migration).

Reply:
ack. Will have this PR without TF native way and a soon to be upcoming PR with the TF native way (awaiting some changes to the install script)